components/eventing-controller/pkg/backend/nats/jetstream/jetstream.go

```diff
@@ -415,6 +427,17 @@ func (js *JetStream) createConsumer(subscription *eventingv1alpha1.Subscription,
 			continue
 		}
 
+		consumerInfo, err := js.jsCtx.ConsumerInfo(js.Config.JSStreamName, jsSubKey.ConsumerName())
```

Comment:
We make too many api calls to the NATS server! Would be nice as a followup to optimize this or find a way around

Reply:
Added a comment, this can be addressed [here](https://github.com/kyma-project/kyma/pull/15295), which is a follow-up to this PR.